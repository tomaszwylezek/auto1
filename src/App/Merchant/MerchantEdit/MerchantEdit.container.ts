import { connect } from 'react-redux';
import { editMerchant } from 'redux/merchants/merchants.actions';
import { MerchantEdit } from './MerchantEdit';

import { getMerchant } from 'redux/merchants/merchants.selector';
import {
  IMerchantEditConnectedActions,
  IMerchantEditConnectedState,
  IMerchantEditProps
} from 'App/Merchant/MerchantEdit/MerchantEdit.types';

const mapStateToProps = (
  state,
  {
    match: {
      params: { id }
    }
  }: IMerchantEditProps
): IMerchantEditConnectedState => ({
  merchant: getMerchant(state, id)
});

const mapDispatchToProps: IMerchantEditConnectedActions = {
  editMerchant
};

export const MerchantEditContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(MerchantEdit);
