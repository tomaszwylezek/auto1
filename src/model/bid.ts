export interface IBid {
  id: string,
  carTitle: string,
  amount: number,
  created: string
}
