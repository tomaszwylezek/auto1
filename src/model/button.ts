export enum ButtonTheme {
  primary = 'primary',
  secondary = 'secondary',
  success = 'success',
  danger = 'danger',
  warning = 'warning',
  info = 'info',
  light = 'light',
  dark = 'dark',
  default = 'default',
  outlinePrimary = 'outline-primary',
  outlineSecondary = 'outline-secondary',
  outlineSuccess = 'outline-success',
  outlineDanger = 'outline-danger',
  outlineWarning = 'outline-warning',
  outlineInfo = 'outline-info',
  outlineLight = 'outline-light',
  outlineDark = 'outline-dark'
}

export enum Size {
  sm = 'sm',
  lg = 'lg'
}
