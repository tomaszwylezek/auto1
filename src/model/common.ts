import { Action } from 'redux';

export interface IAction<T> extends Action {
  payload: T;
}
