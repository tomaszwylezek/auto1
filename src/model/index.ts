export * from './store';
export * from './merchant';
export * from './bid';
export * from './formik';
