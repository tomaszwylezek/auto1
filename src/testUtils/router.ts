import { RouteComponentProps } from 'react-router';
import { UnregisterCallback, Href, Location } from 'history';

export function getMockedRouterProps<P = any>(params?: P, locationData?: Partial<Location>) {
  const location: Location = {
    pathname: '',
    hash: '',
    key: '',
    search: '',
    state: '',
    ...locationData
  };

  const props: RouteComponentProps<P> = {
    match: {
      isExact: true,
      params,
      path: '',
      url: ''
    },
    location,
    history: {
      length: 2,
      action: 'POP',
      location,
      push: jest.fn(),
      replace: () => null,
      go: () => null,
      goBack: jest.fn(),
      goForward: () => null,
      block: () => {
        return null as UnregisterCallback;
      },
      createHref: () => {
        const temp: Href = '';
        return temp as Href;
      },
      listen: () => {
        return null as UnregisterCallback;
      }
    },
    staticContext: {}
  };

  return props;
}
