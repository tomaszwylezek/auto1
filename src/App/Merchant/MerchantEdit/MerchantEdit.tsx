import React, { FC } from 'react';
// Models
import { IMerchantEditProps } from './MerchantEdit.types';

import { IMerchantFormValues } from 'App/Merchant/MerchantForm/MerchantForm.types';
import { MerchantFormContainer } from 'App/Merchant/MerchantForm/MerchantForm.container';
import { Button } from 'shared/Button/Button';
import { ButtonTheme } from 'model';

export const MerchantEdit: FC<IMerchantEditProps> = ({ merchant, editMerchant, history: { push } }) => {
  return (
    <section>
      <Button theme={ButtonTheme.outlineInfo} to="/">
        Back
      </Button>
      <h2>
        Edit merchant {merchant.firstname} {merchant.lastname}
      </h2>
      <MerchantFormContainer
        merchant={merchant}
        onSubmit={(values: IMerchantFormValues) => {
          Promise.resolve(
            editMerchant({
              ...merchant,
              ...values
            })
          ).then(() => {
            push('/');
          });
        }}
      />
    </section>
  );
};
