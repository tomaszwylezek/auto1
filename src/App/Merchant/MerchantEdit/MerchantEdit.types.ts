import { IMerchant } from 'model';
import { RouteComponentProps } from 'react-router';

export interface IMerchantEditOwnProps {}

export interface IMerchantEditConnectedState {
  merchant: IMerchant;
}

export interface IMerchantEditConnectedActions {
  editMerchant: (merchant: IMerchant) => void;
}

export interface IMerchantEditProps
  extends RouteComponentProps<{ id: string }>,
    IMerchantEditConnectedState,
    IMerchantEditConnectedActions,
    IMerchantEditOwnProps {}
