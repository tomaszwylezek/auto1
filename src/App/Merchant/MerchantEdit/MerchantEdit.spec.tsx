import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { MerchantEdit } from './MerchantEdit';
import { getMockedRouterProps } from 'testUtils/router';

// Models
import { IMerchantEditProps } from './MerchantEdit.types';
import { IMerchant } from 'model';

describe('MerchantEdit component should', (): void => {
  let props: IMerchantEditProps;
  let component: ShallowWrapper<IMerchantEditProps>;

  beforeEach((): void => {
    props = {
      editMerchant: jest.fn(),
      merchant: {} as IMerchant,
      ...getMockedRouterProps()
    };

    component = shallow(<MerchantEdit {...props} />);
  });

  it('render correctly', (): void => {
    expect(component).toMatchSnapshot();
  });
});
