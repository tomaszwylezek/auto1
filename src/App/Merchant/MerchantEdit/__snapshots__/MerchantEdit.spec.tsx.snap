// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MerchantEdit component should render correctly 1`] = `
<section>
  <Button
    theme="outline-info"
    to="/"
  >
    Back
  </Button>
  <h2>
    Edit merchant 
     
  </h2>
  <WithFormik(MerchantForm)
    merchant={Object {}}
    onSubmit={[Function]}
  />
</section>
`;
